includes/server/class-llms-rest-memberships-controller.php

```diff
@@ -0,0 +1,590 @@
+			),
+		);
+
+		return array_merge( $parent_links, $links );
+	}
+
+	/**
+	 * Prepare a single object output for response.
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Membership $membership Membership object.
+	 * @param WP_REST_Request $request    Full details about the request.
+	 * @return array
+	 */
+	protected function prepare_object_for_response( $membership, $request ) {
+		$data    = parent::prepare_object_for_response( $membership, $request );
+		$context = $request->get_param( 'context' );
+
+		// Auto enroll.
+		$data['auto_enroll'] = $membership->get_auto_enroll_courses();
+
+		// Catalog visibility.
+		$data['catalog_visibility'] = $membership->get_product()->get_catalog_visibility();
+
+		// Categories.
+		$data['categories'] = $membership->get_categories(
+			array(
+				'fields' => 'ids',
+			)
+		);
+
+		// Instructors.
+		$instructors         = $membership->get_instructors();
+		$instructors         = empty( $instructors ) ? array() : wp_list_pluck( $instructors, 'id' );
+		$data['instructors'] = $instructors;
+
+		// Restriction action.
+		$data['restriction_action'] = $membership->get( 'restriction_redirect_type' );
+		$data['restriction_action'] = $data['restriction_action'] ? $data['restriction_action'] : 'none';
+
+		// Restriction message.
+		$data['restriction_message'] = array(
+			'raw'      => $membership->get( 'restriction_notice', $raw = true ),
+			'rendered' => do_shortcode( $membership->get( 'restriction_notice' ) ),
+		);
+		if ( empty( $data['restriction_message']['raw'] ) ) {
+			$data['restriction_message']['raw']      = 'You must belong to the [lifterlms_membership_link id="' .
+			                                           $membership->get( 'id' ) .
+			                                           '"] membership to access this content.';
+			$data['restriction_message']['rendered'] = do_shortcode( $data['restriction_message']['raw'] );
+		}
+
+		// Restriction page id.
+		if ( 'page' === $data['restriction_action'] || 'edit' === $context ) {
+			$data['restriction_page_id'] = $membership->get( 'redirect_page_id' );
+		}
+
+		// Restriction URL.
+		if ( 'custom' === $data['restriction_action'] || 'edit' === $context ) {
+			$data['restriction_url'] = $membership->get( 'redirect_custom_url' );
+		}
+
+		// Tags.
+		$data['tags'] = $membership->get_tags(
+			array(
+				'fields' => 'ids',
+			)
+		);
+
+		// Sales page type.
+		$data['sales_page_type'] = $membership->get( 'sales_page_content_type' );
+		$data['sales_page_type'] = $data['sales_page_type'] ? $data['sales_page_type'] : 'none';
+
+		// Sales page id.
+		if ( 'page' === $data['sales_page_type'] || 'edit' === $context ) {
+			$data['sales_page_page_id'] = $membership->get( 'sales_page_content_page_id' );
+		}
+
+		// Sales page url.
+		if ( 'custom' === $data['sales_page_type'] || 'edit' === $context ) {
+			$data['sales_page_url'] = $membership->get( 'sales_page_content_url' );
+		}
+
+		return $data;
```

Comment:
Could you please add a filter here?
https://github.com/eri-trabiccolo/lifterlms-rest/blob/memberships/includes/server/class-llms-rest-memberships-controller.php#L354

See #85 (comment)
Basically since then I've planned to add filters/action hooks to the other controllers too.